storage/range.go

```diff
@@ -375,9 +375,13 @@ func (r *Range) redirectOnOrAcquireLeaderLease(timestamp proto.Timestamp) error
 		return r.newNotLeaderError()
 	} else if !held || expired {
 		// Otherwise, if not held by this replica or expired, request renewal.
-		if err := r.requestLeaderLease(timestamp); err != nil {
-			return err
+		err := r.requestLeaderLease(timestamp)
+		// Getting a LeaseRejectedError back means someone else got there
+		// first. Since it's committed by now, we can return a NotLeaderError.
+		if _, ok := err.(*proto.LeaseRejectedError); ok {
```

Comment:
if err can never be nil we should just panic on it. As a reader I'd assume r.requestLeaderLease(timestamp) can return nil.

Its really annoying that golang doesn't allow you to throw invariants all around in the code. modeling it as

if invariant {
panic(..)
}

is ugly